### ISSUE TITLE
Quoting the message often breaks the chat history and messages between quoted message and first 25 messages are permanently disappearing. 

### ISSUE THREAD

**Issue by ekazaev** [GetStream/stream-chat-swift#2527]
## What did you do?

Scrolled chat few pages up. Quoted a message. Closed chat, opened the chat again. Quoted message appears as a part of history and a lot of messages disappeared from the history between the message with the quote and the quoted message itself.

## What did you expect to happen?
History stays as it was before.

## What happened instead?

Quoting the message often breaks the chat history and messages between quoted message and first 25 messages are permanently disappearing. It is possible to fix only after deleting and reinstalling the app. 

## GetStream Environment
**GetStream Chat version:** 4.28.0
**GetStream Chat frameworks:**  GetStream
**iOS version:** 16.3.1
**Swift version:** 5.7
**Xcode version:** Version 14.2 (14C18)
**Device:** Iphone 14 Pro

## Additional context
It is not a new issue. It was reported last year as well and it was able to reproduce withing the stream demo app as well. It was eventually fixed. It also seems that after the fix this issue didnt exist in 2.25 and appears again in 2.26. It looks like some problem in the core data storage as the quoted message is being saved in the same database.

**Comment by nuno-vieira** [comment]
Hi @ekazaev!

Thank you for the report. 

Indeed this issue was introduced in `4.26.0`. But it was recently fixed in `4.28.0`. Are you sure you are using the `4.28.0` version? 

At least we cannot reproduce following your steps in our Demo App. 

Best,
Nuno

**Comment by ekazaev** [comment]
Hi @nuno-vieira 

I am absolutely sure I am using `4.28.0`. And the issue exists both on the sender and receiver side. If I roll back to `4.25.0`, delete the app and reinstall it (to clean core data cache) the issue disappears.

Regards,
Eugene

**Comment by nuno-vieira** [comment]
Hi @ekazaev,

Can you provide your configurations of SDK? At least with the default ones, I cannot reproduce them in our DemoApp. As you can see in the following video:

https://user-images.githubusercontent.com/12814114/225296887-3b91b1dc-4d1c-4eea-99cb-1645c49e75d8.mp4

<br>

Anyway, can you try using our `develop` branch and check if the issue persists on your side? Although this should not happen still on 4.28.0, on `develop` and in the next release, this should be even more stable. 

Best,
Nuno

**Comment by nuno-vieira** [comment]
Hi @ekazaev!

A new 4.29.0 version is out. When you have a chance, can you try this new version and see if the issue is fixed?

Best,
Nuno

**Comment by nuno-vieira** [comment]
Hi @ekazaev!

I'm closing this one for inactivity. But feel free to re-open if you still see the issue.

Best,
Nuno
